### PR TITLE
Master loop emailer code cleanup

### PR DESCRIPTION
**Pull request by lokeshgarg92**

### CODE REVIEW

File: loop/views.py [hunk 1]
```diff
@@ -83,7 +83,8 @@ def dashboard(request):
 
 @csrf_exempt
 def download_data_workbook(request):
-    print request
+    print "hrllo"
+    # print request
```
Comment (sainivikas) [inline loop/views.py]: Please remove debugging statement if not required.

File: loop/utils/emailers_support/queries.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
+__author__ = 'Lokesh'
+
+import MySQLdb
+from dg.settings import *
```
Comment (sainivikas) [inline loop/utils/emailers_support/queries.py]: Please import only things which required.

File: loop/utils/emailers_support/queries.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+
+mysql_cn = MySQLdb.connect(host=DATABASES['default']['HOST'], port=DATABASES['default']['PORT'], user=DATABASES['default']['USER'],
+                           passwd=DATABASES['default']['PASSWORD'],
+                           db=DATABASES['default']['NAME'],
+                           charset='utf8',
+                           use_unicode=True)
+
+
+def onrun_query(query):
+    cursor = mysql_cn.cursor()
```
Comment (sainivikas) [inline loop/utils/emailers_support/queries.py]: use "global variable name" without quotes if using global variables.

File: loop/config.py [hunk 4]
```diff
@@ -9,8 +9,11 @@
 from email import encoders
 from email.header import Header
 from email.mime.base import MIMEBase
+from dg.settings import EXOTEL_ID, EXOTEL_TOKEN, EXOTEL_HELPLINE_NUMBER, MEDIA_ROOT
```
Comment (sainivikas) [inline loop/config.py]: remove variable which not required.

File: loop/config.py [hunk 5]
```diff
@@ -25,7 +28,7 @@ def _create_attachment(self, filename, content, mimetype=None):
         if basetype == 'text':
             encoding = self.encoding or settings.DEFAULT_CHARSET
             attachment = SafeMIMEText(smart_str(content,
-                settings.DEFAULT_CHARSET), subtype, encoding)
+                                                settings.DEFAULT_CHARSET), subtype, encoding)
```
Comment (sainivikas) [inline loop/config.py]: settings not imported anywhere.

File: loop/management/commands/Outliers.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+        # Position of All is first as a co-incidence I think.
+        # Set correct column widths
+        # workbook = create_workbook('Farmer Share Outliers.xlsx')
+        # all_format = ['date_format']
+        # all_format_created = create_format(all_format, workbook)
+
+
+
+        # for aggregator in aggregators:
```
Comment (sainivikas) [inline loop/management/commands/Outliers.py]: remove comments if not required.

File: loop/config.py [hunk 7]
```diff
@@ -1,16 +1,13 @@
...
-from django.template.loader import get_template
 from django.core.mail.message import EmailMultiAlternatives
 import mimetypes
 from email import encoders
 from email.header import Header
 from email.mime.base import MIMEBase
 from django.core.mail.message import DEFAULT_ATTACHMENT_MIME_TYPE
 
+# Code snippet ad imports for email attachment with encodings. To be used in another file for attachmnet in emails
 class EmailMultiAlternativesWithEncoding(EmailMultiAlternatives):
```
Comment (sainivikas) [inline loop/config.py]: why are we extending EmailMultiAlternatives, make comments and reason.

File: loop/management/commands/Outliers.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        parser.add_argument('-td',
+                            dest='to_date',
+                            default=None)
+
+    #generate the excel for the given command line arguments
+    def handle(self, *args, **options):
+        from_to_date = date_setter.set_from_to_date(options.get('from_date'), options.get('to_date'),
+                                                    options.get('num_days'))
+        email_file_list = []
+        if options.get('aggregator') == 'all' or options.get('aggregator') == None:
```
Comment (sainivikas) [inline loop/management/commands/Outliers.py]: Do we need options.get('aggregator') == None condition ?

File: loop/management/commands/Outliers.py [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+            file_to_send = header_dict_for_transport_outlier['workbook_name'] % (
+            MEDIA_ROOT, '', str(from_to_date[0]), str(from_to_date[1]))
+            for elements in header_dict_for_transport_outlier['column_properties']:
+                if 'data_type' in elements.keys() and elements['data_type']=='Date':
+                    date_format = workbook.add_format({'num_format': 'd mmm yy'})
+                    elements['format'] = date_format
+            table_properties = {'data': None, 'autofilter': False, 'banded_rows': False, 'style': 'Table Style Light 15',
+                                'columns': header_dict_for_transport_outlier['column_properties']}
+
+        # Position of All is first as a co-incidence I think.
```
Comment (sainivikas) [inline loop/management/commands/Outliers.py]: remove comments if not required.

File: loop/management/commands/Outliers.py [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+        # all_format_created = create_format(all_format, workbook)
+
+
+
+        # for aggregator in aggregators:
+        #     table_position_to_start = {'row': 2, 'col': 0}
+        #     worksheet_name[aggregator.name_en] = header_dict_for_farmer_outlier['worksheet_name'] % (
+        #     str(aggregator.name_en), str(from_to_date[0]), str(from_to_date[1]))
+
+        table_position_to_start = {'row': 2, 'col': 0}
```
Comment (sainivikas) [inline loop/management/commands/Outliers.py]: if case is not passed then will we need to execute this code ?

File: loop/management/commands/Outliers.py [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+            for elements in header_dict_for_farmer_outlier['column_properties']:
+                if 'data_type' in elements.keys() and elements['data_type']=='Date':
+                    date_format = workbook.add_format({'num_format': 'd mmm yy'})
+                    elements['format'] = date_format
+            table_properties = {'data': None, 'autofilter': False, 'banded_rows': False, 'style': 'Table Style Light 15',
+                                'columns': header_dict_for_farmer_outlier['column_properties']}
+
+
+        if case == 'TransportCost':
```
Comment (sainivikas) [inline loop/management/commands/Outliers.py]: may we use elif, if not then why ?

File: loop/payment_template.py [hunk 12]
```diff
@@ -199,26 +199,23 @@ def excel_processing(workbook, name_of_sheets, heading_of_sheets, heading_format
...
 
-            
     except Exception as e:
         print e
     return workbook
 
 
 def prepare_value_data_generic(data):
-    data = json.loads(data)
+    if type(data) == 'str':
```
Comment (sainivikas) [inline loop/payment_template.py]: What if type(data) is not 'str' then should we proceed to remaining code or terminate here with error message ?

File: loop/utils/emailers_support/date_setter.py [hunk 13]
```diff
@@ -0,0 +1,70 @@
+__author__ = 'Lokesh'
+
+from datetime import *
+from datetime import timedelta
+
+# Enters default value in from_date and to_date
+current_date = None
```
Comment (sainivikas) [inline loop/utils/emailers_support/date_setter.py]: remove variables if not required.

File: loop/utils/emailers_support/queries.py (hunk 3 above)
Comment (lokeshgarg92) [inline loop/utils/emailers_support/queries.py]: not clear

File: loop/management/commands/Outliers.py (hunk 10 above)
Comment (lokeshgarg92) [inline loop/management/commands/Outliers.py]: any of above mentioned IF cases would definitely be passed and no matter which case pass, this line has to be executed.

File: loop/management/commands/Outliers.py (hunk 11 above)
Comment (lokeshgarg92) [inline loop/management/commands/Outliers.py]: yes probably we can. Done

File: loop/payment_template.py (hunk 12 above)
Comment (lokeshgarg92) [inline loop/payment_template.py]: its a patch for the time being.
To answer the question, data parameter would either be String or json object. This If condition would be executed only if data is String else Normal flow will execute

File: loop/utils/emailers_support/queries.py (hunk 3 above)
Comment (sainivikas) [inline loop/utils/emailers_support/queries.py]: no need in this case.

File: loop/payment_template.py (hunk 12 above)
Comment (sainivikas) [inline loop/payment_template.py]: got it.

**Pull request merged**